File: LeetCode/0007-reverse-integer/solution.js
/**
 * @param {number} x
 * @return {number}
 */
var reverse = function(x) {
    let rnum = 0;
    let temp = x;

    while (temp !== 0) {
        let digit = temp % 10;              // extract last digit
        rnum = rnum * 10 + digit;           // build reversed number
        temp = Math.trunc(temp / 10);       // remove last digit
    }

    // 32-bit signed integer range check
    if (rnum < -(2**31) || rnum > (2**31 - 1)) {
        return 0;
    }

    return rnum;
};

File: LeetCode/0007-reverse-integer/solution.java
class Solution {
    public int reverse(int x) {
        int rNum = 0;
        int temp = x;
        while(temp!=0){
            int digit = temp%10;
            if(rNum < Integer.MIN_VALUE / 10 || rNum > Integer.MAX_VALUE / 10){
            return 0;
             }
            rNum = rNum*10 + digit;
            temp = temp/10;
        }
        
        return rNum;
    }
}
